src/xform_to_datalog/arcs_manifest_tree/handle_connection_spec.h

```diff
@@ -34,18 +36,45 @@ class HandleConnectionSpec {
         : (reads_) ? arcs::HandleConnectionSpecProto_Direction_READS
                    : arcs::HandleConnectionSpecProto_Direction_WRITES);
     result.set_name(name_);
+    *result.mutable_type() = type_->MakeProto();
     return result;
   }
 
+  // Get all of the AccessPaths rooted at the associated ParticleSpec
+  // (indicated through the provided name) through leaf fields of type_.
+  std::vector<raksha::ir::AccessPath> GetAccessPaths(
+      std::string particle_spec_name) const {
+    namespace ir = raksha::ir;
+    std::vector<ir::AccessPath> result_paths;
+    ir::AccessPathSelectorsSet selectors_set =
```

Comment:
Is a copy needed here?

Reply:
There isn't actually a copy that happens here. `GetAccessPathSelectorsSet()` returns an `AccessPathSelectorsSet` that lives in a local by value, so inside the function the copy is elided. At the callsite (here), we are just providing a local name to which we can attach the value. As such, if I understand C++17 correctly, no copies occur.